telegram/src/main/kotlin/com/github/kotlintelegrambot/entities/ChatId.kt

```diff
@@ -4,9 +4,13 @@ package com.github.kotlintelegrambot.entities
  * Unique identifier for the target chat or username of the target channel (in the format @channelusername)
  */
 sealed class ChatId {
-    data class Id(val id: Long) : ChatId()
+    data class Id(val id: Long) : ChatId() {
+        override fun toString() = id.toString()
+    }
+
     class ChannelUsername(username: String) : ChatId() {
         val username: String = if (username.startsWith("@")) username else "@$username"
+        override fun toString() = username
     }
```

Comment:
Hmm, I'm not a sure about relying on the `toString` method to transform a `ChatId` for multipart body requests. 

What do you think about having an extension function like the next encapsulating that transformation instead:

```kotlin
fun ChatId.toMultipartBodyPart(partName: String): MultipartBody.Part = when(this) {
    is Id -> id.toMultipartBodyPart(partName)
    is ChannelUsername -> username.toMultipartBodyPart(partName)
}
```

We can add it in the `MultipartExtensions.kt` file so it's grouped with other similar functions.

Reply:
Reason why I don't like such extension method usage: 
converting to MultipartBody is totally none of the business of ChatId class, so it shouldn't be a part of this class. Also creating extension methods decreases code readability, because instead of looking in 1 file with class definition, developer should looks in multiple files (which he should find first).

`toString()` here is doing exactly what is was made for: returns a string representation of the object. At the end of the day, method with converts object to string RequestBody expects string representation of the object as input, which is `toString()`

Reply:
The extension function I was proposing in the comment is not a thing of `ChatId` class. I don't want it to be part of the class, that's why I'm proposing the implementation with an extension function and not with a member function. Using extension functions to map a class to a different representation of that class is a common pattern in Kotlin. I don't see how that decreases code readability. If we had to implement the same without extension functions we'll probably create a class with a single function that will receive the `ChatId` as parameter and will return a `String`, and that will be harder to read and use in an IDE than the extension function.

`toString()` says that the returned string should be "a concise but informative representation that is easy for a person to read", and that's true with your changes. Why I'm not in favour of using `toString` for this then?:

* If this value returned by `toString()` is used when we need to map the chat id value for multipart parts and clients widely use it, we'll be stuck with it, and if we change it we'll break all these clients. In this case we're the only clients supposed to do this, so it's not such a big deal, but we'll be limiting ourselves to be stuck with this `toString` implementation.
* If we wanted a different human readable representation returned by `toString` for this class for debugging purposes, we couldn't have it because we would break the multipart code.
* Currently, we don't override the `toString()` for other Telegram Bot Api classes and rely on the `toString` generated by data classes (actually there are some classes that override it with the same idea you're proposing here too, but we want to move them away from that). So it won't be consistent with the `toString` of other classes representing Telegram Bot API objects, which is not a big deal either, but could be annoying to library clients.
* If in the future we need to add a new subclass for the `ChatId` sealed class and we forget to override the `toString`, our code will silently fail. If we do that and we're using the extension function approach, the code won't compile until we consider the new subclass in our mapping code.

What do you think about this?

Reply:
Alright, your arguments are quiet reasonable, although I found a better way of doing it 🙃 
`ChatIdConverterFactory` now implements `requestBodyConverter()` which converts class to `RequestBody` under `@Part` annotation in `@Multipart` requests.